wallets/provider-walletconnect-2/src/index.ts

```diff
@@ -106,23 +107,9 @@ export const connect: Connect = async ({ instance, network, meta }) => {
   });
   // Override the value (session).
   instance.session = session;
-  const currentChainId = !isNew
-    ? String(await getPersistedChainId(instance))
-    : undefined;
-  const accounts = getAccountsFromSession(session, currentChainId);
-  /*
-   * TODO: we need to fix next lines to support multiple accounts
-   * for now, it will return the current evm account on the current chain
-   */
-  if (!isNew) {
-    return {
-      chainId: String(currentChainId),
-      accounts: accounts?.[0].accounts,
-    };
-  }
-  const newChainId = accounts?.[accounts.length - 1].chainId;
-  void persistCurrentChainId(instance, newChainId);
-  return accounts?.[accounts.length - 1];
+  let currentChainId = !isNew ? await getPersistedChainId(instance) : undefined;
```

Comment:
`isNew` is using here only, couldn't we call `await getPersistedChainId(instance)` anyway and fallback to `false` whenever something goes wrong?
If yes, we can remove whole `isNew` stuff to simplify the code.

Reply:
I'm not sure what you mean but we need to only use persisted chain id in case of reconnect to set correct state for our wallet core.

Reply:
I mean if failing `getPersistedChainId` should reject the whole `connect` (it only gets chainId from storage), we can do something like this:

session.ts
```ts
export async function getPersistedChainId(instance: any) {
  try {
    return (await instance.client.core.storage.getItem(CHAIN_ID_STORAGE))
      ?.defaultChainId;
  } catch {
    return undefined;
  }
}
```

index.ts
```ts
let currentChainId = await getPersistedChainId(instance);
```

and then `isNew` logic codes since the variable is not used anymore.

Reply:
I think it's not correct because we might have some stale values for default chain id in storage (from previous connections) which shouldn't be used in case of a new connection.

Reply:
Here is how I see the flow:

`tryConnect` will try to restore the session, if it couldn't (stale) it will call `disconnectSessions` which remove old values.
if there is no session, it will create a new session and set `isNew = true`.

afterwards, we have this code:
```
  let currentChainId = !isNew ? await getPersistedChainId(instance) : undefined;
  currentChainId = currentChainId ? String(currentChainId) : undefined;
```
which means if a session restored, try get the chain from session. (it not possible to session be staled here, as explained before).
So we can safely read from storage and if the value is not there it will return `undefined`

Reply:
thanks for the explanation. I've changed the code now.